internal/service/apigateway/rest_api.go

```diff
@@ -55,7 +55,10 @@ func ResourceRestAPI() *schema.Resource {
 			},
 			"created_date": {
 				Type:     schema.TypeString,
-				Computed: true,
+
+			"fail_on_warnings": {
```

Comment:
It's important to note that the [failOnWarnings](https://docs.aws.amazon.com/apigateway/latest/api/API_PutRestApi.html) parameter only applies to AWS's PutRestApi() method when using an OpenAPI definition. This is an update of an existing API only.

Terraform's implementation of this resource is slightly divergent from the intended use of the API where Terraform first creates a REST API (without an OpenAPI definition) and then if a OpenAPI definition is supplied immediately updates the API using the OpenAPI definition.